File: leetcode136.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        dic = {}
        for x in nums:
            dic[x] = dic.get(x,0) + 1
            if dic[x] == 2:
                dic.pop(x)
        return list(dic.keys())[0]

File: leetcodec++/leetcode136.cpp
//
// Created by 黄敏辉 on 2019-08-12.
//

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int res = 0;
        for (int i = 0; i < nums.size(); i++)
            res ^= nums[i];
        return res;
    }
};